### PR TITLE
System requirements specification document

### PR DESCRIPTION
**Pull request by TemirlanAidarov**
## Proposed changes

[//]: # The system requirements specification(SRS) document is added. 
Basically, SRS is a document that describes what the software will do and how it will be expected to perform. It also describes the functionality the product needs to fulfill all stakeholders needs.

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

**Comment by christian-bromann** [comment]
Hey @TemirlanAidarov thanks for raising the PR. I have two questions:

1. What was your motivation writing up this document?
2. I see a lot of duplicate content here that we already put into `PROJECT_CHARTER.md` and `ROADMAP.md`. I would like to avoid to have to maintain this in multiple places so I am curious if there is a need to have a specific System requirements specification document or if we can just use references to existing file.

**Comment by BorisOsipov** [comment]
@christian-bromann I had same question. the answer seems:
```
I am Computer Science student who takes a course related to Open Software Project.
We have an assignment to write/improve System Requirements Specification (SRS) document for any project. 
```

https://github.com/eslint/eslint/discussions/15637#discussion-3900130

**Comment by TemirlanAidarov** [comment]
@christian-bromann thank you for your feedback. 
**1.** "What was your motivation writing up this document"
A System Requirements Specification (SRS) is a document that describes the features and behavior of a system or software application. Also, SRS is the plan of action and keeps all team members on the same page. Also, I think it will be a good learning material for anyone interested in WebDriverIO.

Also, as Boris Osipov said, it was my university assignment. However, I got very passionate about my writing and I became very enthusiastic about the WebDriverIO project and really tried to make a contribution. The previously submitted document was a raw draft. I have made significant changes and reorganized the document. Also, I tried to add illustrations and other useful information. Now I think it should look much better. Can you check for the newer version, please?

**2.** Initially, I thought to gather all the needed information from other documentation in one file. But after your comment, I have realized that it would be convenient if there are links to other documentation. So I have changed and restructured my document.

File: SRS Document.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+* Chrome Dev Tools Protocol (for chromium based automation)
+* Appium (for mobile automation)
+
+## 2.3 User Classes and Characteristics
+In general, WebdriverIO is characterized as a simple and accessible tool for creating web and mobile app tests. It is accessible to anyone who wants to learn and write frontend tests in JavaScript (TypeScript) and needs some assistance.
+
+WebdriverIO is easy to start and get used to for beginners. The installation process is straightforward and quick. Users do not need to install any browser drivers and configure themselves. After installation, the user only needs to write a config command that will perform all the necessary configuration. Further details are described in the [documentation page](https://webdriver.io/docs/gettingstarted) 
+
+## 2.4 Operating Environment
+A user has to install Node.js. It is suggested to choose the *LTS* version, so it is suggested to choose 12.16.1 or above.
```
Comment (christian-bromann) [inline SRS Document.md:80]: We will deprecated v12 soon so I'd suggest to change the wording and say "WebdriverIO suggests to use LTS and is compatible with other Node.js versions that are in maintenance state" ... something like that.

File: SRS Document.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+In general, WebdriverIO is characterized as a simple and accessible tool for creating web and mobile app tests. It is accessible to anyone who wants to learn and write frontend tests in JavaScript (TypeScript) and needs some assistance.
+
+WebdriverIO is easy to start and get used to for beginners. The installation process is straightforward and quick. Users do not need to install any browser drivers and configure themselves. After installation, the user only needs to write a config command that will perform all the necessary configuration. Further details are described in the [documentation page](https://webdriver.io/docs/gettingstarted) 
+
+## 2.4 Operating Environment
+A user has to install Node.js. It is suggested to choose the *LTS* version, so it is suggested to choose 12.16.1 or above. 
+There are no other strict rules regarding system requirements. Nevertheless, in order to work with little problem 
+as possible with Node.js and WebdriverIO, it is advised for users to have:
+* Ubuntu version 16.04 or higher for Linux 
+* or Windows 7 or newer versions with at least 2GB RAM (4GB preferable).
```
Comment (christian-bromann) [inline SRS Document.md:84]: I don't see how systems not listed here (e.g. Mac) should have more problems to run WebdriverIO. We test this software on all broader operating systems.

File: SRS Document.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+## 2.4 Operating Environment
+A user has to install Node.js. It is suggested to choose the *LTS* version, so it is suggested to choose 12.16.1 or above. 
+There are no other strict rules regarding system requirements. Nevertheless, in order to work with little problem 
+as possible with Node.js and WebdriverIO, it is advised for users to have:
+* Ubuntu version 16.04 or higher for Linux 
+* or Windows 7 or newer versions with at least 2GB RAM (4GB preferable). 
+
+## 2.5 Assumptions and Dependencies
+The WebdriverIO community has to build a full test framework with various additional features (e.g. making the CLI tool more powerful, Jest Framework Support). Still, the WebdriverIO community is active and working on continuous improvements.
```
Comment (christian-bromann) [inline SRS Document.md:87]: We don't support the Jest framework (note that you can run WebdriverIO in all frameworks but the testrunner only supports Mocha, Jasmine and Cucumber as test frameworks). We allow however to write tests in the same way as Jest tests are written.

File: SRS Document.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+## 3.2 Software quality attributes
+* Software should be user friendly
+* Software should be open source, where everyone can offer a new idea and contribute in accordance with the rules.
+* Software should have accessible product support
+* Software should have good documentation and tutorials
+* Software should support a variety of community plugins
+* Software should be integrable and extensible to be customized according to the user's need.
+
+## 3.3 Software quality attributes
+TBD
```
Comment (christian-bromann) [inline SRS Document.md:112]: Any plans to continue this document? Which sections are missing here?

**Review by christian-bromann — COMMENTED** [review]
Sorry for the late in depth review but I was sort of on vacation last week.

There are some terminology issues and some wrong assumptions in the document. Happy to work it out with you if you are interested.

**Comment by christian-bromann** [comment]
@TemirlanAidarov it doesn't seem this PR is complete. Is there still interest continuing in this effort?

**Comment by christian-bromann** [comment]
Closing due to inactivity.

**Pull request closed without merging**